app/controllers/likes_controller.rb

```diff
@@ -0,0 +1,38 @@
+class LikesController < ApplicationController
+  include ActionView::RecordIdentifier
+  def update
+    if likes.exists?
+      likes.destroy_all
+    elsif current_user.likes.count < User::LIKE_LIMIT
+      likes.first_or_create(user: current_user)
+    else
+      errors = "You've reached the like limit"
+    end
+
+    turbo_streams = [
+      turbo_stream.replace("top-card", partial: "posts/top_card", locals: { name: current_user.name, remaining_likes_quantity: User::LIKE_LIMIT - current_user.likes.count, errors: errors } ),
+      turbo_stream.replace(dom_id(resource), partial: "posts/#{resource_name}", locals: { "#{resource_name}".to_sym => resource })
+    ]
```

Comment:
For this assignment, I liked that I was able to use the same controller to like posts and comments, but I'd try to have a conversation asking if the approach is too hacky and what's the maintainability and scalability of this based on possible future requirements. 

One problem for example is, are all the `resource` partials going to live under the posts folder?
How does this work if we try to like something different than a post, or something that is nested under a different model

Reply:
We have a pretty good solution for this in Strety in which you create a base controller where you define all the main logic. And in the controllers that inherit it you can override the `set_resource` method for example depending on which resource you are liking. For example:

```
/dogs/:id/likes
```
would have a before_action like:
```
def set_resource
    @resource = Dogs.find(params[:id])
end
```

Good thinking and questioning on this!